### ISSUE TITLE
Question on reproduce the work

### ISSUE THREAD

**Issue by Pokemon-Huang** [TuSimple/sparse-structure-selection#3]
Excuse me, I 've trained Resnet101 on CIFAR10 with your Mxnet code, But I barely got zero lambda_block, is this normal?
Here is my lambda_block, almost all lamda is bigger than 1,
[-1.22, -1.32, 1.35, -1.31, -8.30, -1.59, 1.63, -1.42, 1.19, 1.37, ... 1.77, 1.78, 1.81, 1.72, 1.14, -0.09, 0.77]
BTW, I trained network for 100 epoch without changing gamma, learning rate & other config, but my environment is mxnet 1.5.0, cuda9.0, ubuntu16.04

**Comment by huangzehao** [comment]
@Pokemon-Huang 
Sorry for the late reply.
Hi, it seems that the APGNAG doesn't work correctly because of the different mxnet version. In the latest mxnet, the solver will call update_multi_precision instead of update. 
Add the following codes in https://github.com/TuSimple/sparse-structure-selection/blob/master/core/optimizer.py may solve this problem:
```
def update_multi_precision(self, index, weight, grad, state):
    return self.update(index, weight, grad, state)
```

**Comment by Pokemon-Huang** [comment]
thank u,  it solved my problem perfectly.
